packages/protobuf/src/codegenv1/scalar.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ScalarType } from "../descriptors.js";
+
+/**
+ * Return the TypeScript type (as a string) for the given scalar type.
+ */
+export function scalarTypeScriptType(
```

Comment:
This function moves from `@bufbuild/protobuf/reflect` to `@bufbuild/protobuf/codegenv1`. IMO it's a much better fit because the function is intended to be used for generating code.